### ISSUE TITLE
rm <symlink> removes target instead

### ISSUE THREAD

**Issue by ne-on** [nushell/nushell#2350]
The following steps:

```sh
touch test
ln -s test t
rm t
```

cause the file `test` to be removed instead of the link `t`.

Subsequent calls to `rm t` produce an error:

```sh
error: No valid paths
  ┌─ shell:1:1
  │
1 │ rm t
  │ ^^ no valid paths
```

Running macOS 10.14, nushell 0.18.1 installed from brew.

**Comment by JosephTLyons** [comment]
Hmm, I'm not experiencing this.  When I run `rm t`, it removes the symlink, not the `test` file.  I'm on macOS 10.15.6.

**Comment by thegedge** [comment]
☝️ same for me, both from master and Homebrew.

@ne-on can you verify? If it's actually deleting the target instead of the symlink, can you think of any other reasons why that may be happening?

**Comment by ne-on** [comment]
I went through the config file commenting things out.
I only seem to have the problem when `rm_always_trash = true`

**Comment by thegedge** [comment]
Thanks, @ne-on . It's possible it's a bug in the crate we use to send files to the trash (https://crates.io/crates/trash). If so, we can keep this issue open until we fix it upstream and bump the version in nushell.

**Comment by ne-on** [comment]
Makes sense. Just glad to help, @thegedge, even if small. I really like what you all are doing here. Thanks.

**Comment by SanchithHegde** [comment]
@ne-on I tried to reproduce this issue with the current main branch (commit hash 2c89a228d54ab70b25caecdeeb2c2d4daf145e6d) with the config variable `rm_always_trash` set to `true` (that's the only variable in my config), but it's deleting the symlink as expected. Could you please test it again on main or maybe provide more info/steps to reproduce this?

I'm running Arch Linux, if that matters.

**Comment by SanchithHegde** [comment]
After some more digging, I found that this has been fixed in `trash 1.2.0` (see https://github.com/ArturKovacs/trash-rs/issues/22) and currently, `trash 1.3.0` is used in the main branch. If @ne-on can confirm that this is fixed, I believe this issue can be closed.
